workflow/rules/sra_download.smk

```diff
@@ -0,0 +1,108 @@
+"""Independent Snakefile for downloading samples from SRA."""
+
+import pandas as pd
+
+samples = pd.read_csv(config["samples"], header=0, index_col=0, sep = "\t")
+DOWNLOAD_DIR = config["outdir"]
+# Write fastq.gz location into new sample table.
+SAMPLES_OUT = config["samples_out"]
+samples_mod = samples.copy()
+
+localrules: prefetch, add_fq_file_path, all
+
+rule all:
+    input:
+        SAMPLES_OUT
+
+
+rule prefetch:
+    "Prefetch SRA entry. Requires internet access."
+    output:
+        os.path.join(DOWNLOAD_DIR, "{sample}", "{sample}.sra")
+    params:
+        outdir = DOWNLOAD_DIR
+    conda:
+        os.path.join(workflow.basedir, "..", "envs", "sra-tools.yaml")
+    singularity:
+        "docker://ncbi/sra-tools"
+    shell:
+        """
+        prefetch {wildcards.sample} --output-directory {params.outdir}
+        """
+
+rule fasterq_dump:
+    input:
+        os.path.join(DOWNLOAD_DIR, "{sample}", "{sample}.sra")
+    output:
+        os.path.join(DOWNLOAD_DIR, "{sample}", "{sample}.dumped")
+    params:
+        outdir = lambda wildcards: os.path.join(DOWNLOAD_DIR, wildcards.sample),
+        cluster_log_path = config["cluster_log_dir"]
+    resources:
+        mem_mb = 2000
```

Comment:
maybe use this: mem_mb=lambda wildcards, attempt: 2000 * attempt
 which would increase the memory with each attempt (I am not sure how much of a problem this could be or why fasterq-dump requires this.